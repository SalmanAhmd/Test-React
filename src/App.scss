body {
    margin: 0;
    padding: 0;
    // background-color: #707174;
}

.App{
    padding: 4em;
    margin: auto;
    text-align: center;
}
