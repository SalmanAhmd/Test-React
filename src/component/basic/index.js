import React from 'react';

export const Basic = () => {
  return (
    <div >
      <h1> Basic Test</h1>
      <p> This is a basic Test Component</p>
    </div>
  );
}

export default Basic;
