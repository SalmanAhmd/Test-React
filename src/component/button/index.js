import React from 'react';
import './button.scss'

export const Button = ({ label }) => {

  return (
    <div data-testid='button' className='button-style' >Hi {label}</div>
  )
}

export default Button;
