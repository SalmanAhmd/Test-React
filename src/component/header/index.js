import React from 'react';
import './style.scss';
import Logo from '../../assets/img/logo.png'

const Header = () => {
    return (
        <header>
            <div className='wrap'>
                <div className='logo'>
                    <img src={Logo} alt='Logo' />
                </div>
            </div>
        </header>
    )
}

export default Header;
