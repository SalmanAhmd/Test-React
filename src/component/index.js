export * from './button';
export * from './link';
export * from './account';
export * from './basic';