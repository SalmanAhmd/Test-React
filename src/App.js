import React, { useState, useEffect } from 'react';

import { Account, Button, TestHook, TestHookContext } from './component'
import './App.scss';
import Context from './store/context';


function App() {

  const [state, setState] = useState({
    isLoading: true,
    users: [],
    error: null,
    home: "home testing"
  });

  useEffect(() => {
    fetch(`https://jsonplaceholder.typicode.com/users`)
      .then((response) => response.json())
      .then((data) =>
        setState(prev => ({
          ...prev,
          users: data,
          isLoading: false,
        })))
      .catch((error) => setState(prev => ({
        ...prev,
        error,
        isLoading: false
      })));
  }, [setState])


  const [state1, setState1] = useState("Some Text");
  const [name, setName] = useState("Moe");

  const changeName = () => {
    setName("Steve")
  }
  const changeText = () => {
    setState1("Some Other Text")
  }


  const { isLoading, users, error } = state;

  return (
    <>
      <h1>Display Active Users Account Details</h1>
      {error ? <p>{error.message}</p> : null}
      {!isLoading ? (
        users.map((user) => {
          return <Account key={user.username} user={user} />;
        })
      ) : (
          <h3>Fetching Users...</h3>
        )}
      <Button label='button' />

      {/* -------------------- */}
      <h1> Basic Hook useState </h1>
      <TestHook name={name} changeName={changeName} />
      <h1> Basic Hook useContext</h1>
      <Context.Provider value={{
        changeTextProp: changeText,
        stateProp: state1
      }} >
        <TestHookContext />
      </Context.Provider>
    </>
  );
}

export default App;
