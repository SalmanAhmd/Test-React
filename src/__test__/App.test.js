import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";

import App from "../App";
import { Account } from '../component'

// skip test
it.skip("renders without crashing", () => {
  shallow(<App />);
});

const user = {
  name: "Adeneye David",
  email: "david@gmail.com",
  username: "Dave",
};


it("renders Account header", () => {
  const wrapper = shallow(<App />);
  const welcome = <h1>Display Active Users Account Details</h1>;
  expect(wrapper.contains(welcome)).toEqual(true);
});

describe("mock user name", () => {
  it("accepts user account props", () => {
    const wrapper = mount(<Account user={user} />);
    expect(wrapper.props().user).toEqual(user);
    wrapper.unmount();
  });
  it("contains users account email", () => {
    const wrapper = mount(<Account user={user} />);
    const value = wrapper.find("p").text();
    expect(value).toEqual(user.email);
    wrapper.unmount();
  });
  // it("renders correctly with no error message", () => {
  //   const wrapper = mount(<Account user={user} />);
  //   expect(wrapper.state("error")).toEqual(null);
  // });
});

it("renders correctly Shallow", () => {
  const treeShollow = shallow(<App />);
  expect(toJson(treeShollow)).toMatchSnapshot();
});

it("renders correctly Mount", () => {
  const treeMount = mount(<App />);
  expect(toJson(treeMount)).toMatchSnapshot();
  treeMount.unmount();
});
