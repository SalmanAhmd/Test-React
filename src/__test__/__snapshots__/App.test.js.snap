// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Mount 1`] = `
<App>
  <h1>
    Display Active Users Account Details
  </h1>
  <h3>
    Fetching Users...
  </h3>
  <Button
    label="button"
  >
    <div
      className="button-style"
      data-testid="button"
    >
      Hi 
      button
    </div>
  </Button>
  <h1>
     Basic Hook useState 
  </h1>
  <TestHook
    changeName={[Function]}
    name="Moe"
  >
    <div>
      <button
        onClick={[Function]}
      >
        State Change Button
      </button>
      <p>
        Initial State
      </p>
      <button
        onClick={[Function]}
      >
        Change Name
      </button>
      <p>
        Moe
      </p>
    </div>
  </TestHook>
  <h1>
     Basic Hook useContext
  </h1>
  <TestHookContext>
    <div>
      <button
        onClick={[Function]}
      >
        Change Text
      </button>
      <p>
        Some Text
      </p>
    </div>
  </TestHookContext>
</App>
`;

exports[`renders correctly Shallow 1`] = `
<Fragment>
  <h1>
    Display Active Users Account Details
  </h1>
  <h3>
    Fetching Users...
  </h3>
  <Button
    label="button"
  />
  <h1>
     Basic Hook useState 
  </h1>
  <TestHook
    changeName={[Function]}
    name="Moe"
  />
  <h1>
     Basic Hook useContext
  </h1>
  <ContextProvider
    value={
      Object {
        "changeTextProp": [Function],
        "stateProp": "Some Text",
      }
    }
  >
    <TestHookContext />
  </ContextProvider>
</Fragment>
`;
